symphony/Symphony/Plugins/MediaStreamPicker/MediaStreamPickerViewModel.cs

```diff
@@ -149,48 +156,91 @@ void _onTimer(object state)
 
         void rebuild()
         {
-            streams.Clear();
+            screenStreams.Clear();
+            windowStreams.Clear();
             mediaStreams.Clear();
 
             // Note:
             // :0 needed on the end in chromium 49+
             // see discussion here: https://bitbucket.org/chromiumembedded/cef/issues/1065/add-support-for-webrtc-based-screen
             foreach (EnumScreenResult screen in screens)
             {
-                addToStreams(screen.title, screen.image, "fullscreen");
+                addToScreenStreams(screen.title, screen.image, "fullscreen");
                 mediaStreams.Add("screen:" + screen.id + ":0");
             }
             foreach (EnumWindowResult window in windows)
             {
-                addToStreams(window.title, window.image, window.filename.ToString());
+                addToWindowStreams(window.title, window.image, window.filename.ToString());
                 mediaStreams.Add("window:" + window.hWnd.ToString() + ":0");
             }
 
-            Streams = streams;
+            ScreenStreams = screenStreams;
+            WindowStreams = windowStreams;
+
             selectedIndex = -1;
-        }
-
-        void addToStreams(string title, BitmapSource image, string fileName)
+        }
+
+        void addToScreenStreams(string title, BitmapSource image, string fileName)
         {
             Img item = new Img(title, image, fileName);
-            streams.Add(item);
-        }
-
+            screenStreams.Add(item);
+        }
+
+        void addToWindowStreams(string title, BitmapSource image, string fileName)
+        {
+            Img item = new Img(title, image, fileName);
+            windowStreams.Add(item);
+        }
+
         RequestShareEventArgs getSelectedMediaStream()
         {
-            if (selectedIndex == -1 || selectedIndex < 0 || selectedIndex >= mediaStreams.Count)
-                return null;
-            RequestShareEventArgs args = new RequestShareEventArgs(mediaStreams[selectedIndex], streams[selectedIndex].fileName, streams[selectedIndex].Str);
+            if (selectedIndex == -1 || selectedIndex < 0 || (selectedIndex + screenStreams.Count >= mediaStreams.Count))
+                return null;
+            
+            RequestShareEventArgs args = null;
+
+            //If both Applications & Screens are enabled.
+            if (Sources.Contains("screen") && Sources.Contains("window")) {
```

Comment:
could below looks like it is repeating itself a lot... e.g., 205 looks really similar to 220, 209 looks similar to 220.